src/smart-components/dialog-routes/portfolio-routes.js

```diff
@@ -55,21 +59,33 @@ const AddPortfolioModal = lazy(() =>
 
 const PortfolioRoutes = () => {
   const viewState = useInitialUriHash();
-  const { portfolioItemId, portfolioItemUserCapabilities } = useSelector(
-    (state) => ({
-      portfolioItemId:
-        state?.portfolioReducer?.portfolioItem?.portfolioItem?.id,
-      portfolioItemUserCapabilities:
-        state?.portfolioReducer?.portfolioItem?.portfolioItem?.metadata
-          ?.user_capabilities
-    })
-  );
+  const {
+    portfolioItemId,
+    portfolioItemUserCapabilities,
+    portfolios,
+    selectedPortfolio
+  } = useSelector((state) => ({
```

Comment:
this selector is now probably updated all the time and re-rendering the whole thing. Could you split it into one selector per value and add shallow equal?